### ISSUE TITLE
StackOverflow Question: Facenet/OpenFace realtime clustering

### ISSUE THREAD

**Issue by angelorodem** [davidsandberg/facenet#695]
The question below was made by me on stackoverflow, can you help?

Recently i got interested in facial recognition and i read that that the OpenFace and Facenet fit a face in a 128-dimension sphere, and images of the same face tend to stay close to each other, in Openface docs they wrote about clustering and Facenet there is a python script related to clustering, but there very few explanations about how the whole clustering works.

Is it possible from a face detected from a live stream, fit it in the 128-d sphere, analyze to which cluster it belongs, being that this cluster is being populated in real-time as the photos are being fitted in the sphere, and is there some easy to understand example?

Thanks.

**Comment by speculaas** [comment]
Hi, Angelorodem, 
I think maybe 
https://github.com/davidsandberg/facenet/blob/master/contributed/cluster.py
can help?

**Comment by khadijakhaldi** [comment]
did you figure out the answer ?

**Comment by angelorodem** [comment]
nope, i ended up with a solution that gave a incredible number of false positives, so i'm still waiting
